sdk/wallet-standard/src/wallet.ts

```diff
@@ -10,6 +22,95 @@ declare module '@wallet-standard/core' {
 		 */
 		readonly id?: string;
 	}
+
+	export interface StandardConnectOutput {
+		supportedIntents?: string[];
+	}
 }
 
 export type { Wallet } from '@wallet-standard/core';
+
+export function getNormalizedSuiWallets<
+	AdditionalFeatures extends keyof KnownFeatures & keyof Wallet['features'] = never,
+	KnownFeatures extends Wallet['features'] = SuiWalletFeatures,
+>(
+	features: AdditionalFeatures[] = [],
+): WalletWithFeatures<Pick<KnownFeatures, AdditionalFeatures | keyof MinimallyRequiredFeatures>>[] {
+	const wallets = getWallets().get();
+
+	return wallets
+		.map((wallet) => normalizeWalletFeatures(wallet))
+		.filter((wallet) => isWalletWithRequiredFeatureSet(wallet, features)) as WalletWithFeatures<
+		Pick<KnownFeatures, AdditionalFeatures | keyof MinimallyRequiredFeatures>
+	>[];
+}
+
+function normalizeWalletFeatures(wallet: WalletWithFeatures<Partial<SuiWalletFeatures>>) {
+	const features = {
+		...wallet.features,
+	};
+
+	if (
+		wallet.features['sui:signTransactionBlock'] &&
+		!wallet.features['sui:signTransactionBlock:v2']
+	) {
+		const { signTransactionBlock } = wallet.features['sui:signTransactionBlock'];
+		features['sui:signTransactionBlock:v2'] = {
+			version: '2.0.0',
+			signTransactionBlock: async (input) => {
+				const transactionBlock = TransactionBlock.from(input.transactionBlock);
+				const { transactionBlockBytes, signature } = await signTransactionBlock({
+					...input,
+					transactionBlock,
+				});
+
+				return { transactionBlockBytes, signature };
+			},
+		};
+	}
+
+	if (
+		wallet.features['sui:signAndExecuteTransactionBlock'] &&
+		!wallet.features['sui:signAndExecuteTransactionBlock:v2']
+	) {
+		const { signAndExecuteTransactionBlock } =
+			wallet.features['sui:signAndExecuteTransactionBlock'];
+		features['sui:signAndExecuteTransactionBlock:v2'] = {
+			version: '2.0.0',
+			signAndExecuteTransactionBlock: async (input) => {
+				const transactionBlock = TransactionBlock.from(input.transactionBlock);
+				const { digest, rawEffects, balanceChanges } = await signAndExecuteTransactionBlock({
+					...input,
+					transactionBlock,
+					options: {
+						showRawEffects: true,
+						showBalanceChanges: true,
+						showRawInput: true,
+					},
+				});
+
+				return {
+					digest,
+					effects: rawEffects ? toB64(new Uint8Array(rawEffects)) : null,
+					balanceChanges:
+						balanceChanges?.map(({ coinType, amount, owner }) => {
+							const address =
+								(owner as Extract<typeof owner, { AddressOwner: unknown }>).AddressOwner ??
+								(owner as Extract<typeof owner, { ObjectOwner: unknown }>).ObjectOwner;
+
+							return {
+								coinType,
+								amount,
+								address,
+							};
+						}) ?? null,
+				};
+			},
+		};
+	}
+
+	return {
+		...wallet,
```

Comment:
I'm curious if this is guaranteed to be safe. These wallets could be classes, so I wonder if we should avoid doing this for any reason?

Reply:
Non of the properties are methods, but it's possible they are defined as getters that wouldn't be spread.  I think doing a more explicit clone rather than a spread should be safe?